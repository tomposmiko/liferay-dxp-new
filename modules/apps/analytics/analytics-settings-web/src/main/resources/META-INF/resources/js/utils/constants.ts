/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

const IMAGES_PATH = Liferay.ThemeDisplay.getPathThemeImages();

export const SPRITEMAP = IMAGES_PATH + '/clay/icons.svg';

export const NOT_FOUND_GIF = `${IMAGES_PATH}/states/search_state.gif`;

export const EMPTY_STATE_GIF = `${IMAGES_PATH}/states/empty_state.gif`;

export const ERROR_MESSAGE = Liferay.Language.get(
	'an-unexpected-system-error-occurred'
);
