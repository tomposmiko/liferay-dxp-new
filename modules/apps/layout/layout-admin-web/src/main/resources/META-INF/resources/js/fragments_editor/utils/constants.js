/**
 * Available element config keys
 */
export const ITEM_CONFIG_KEYS = {
	backgroundColor: 'backgroundColor',
	columnSpacing: 'columnSpacing',
	containerType: 'containerType',
	padding: 'padding'
};

/**
 * Available values for containerType config key
 */
export const CONTAINER_TYPES = [
	{
		containerTypeId: 'fluid',
		label: Liferay.Language.get('fluid')
	},

	{
		containerTypeId: 'fixed',
		label: Liferay.Language.get('fixed-width')
	}
];

/**
 * Fragments Editor item borders
 * @review
 * @type {!object}
 */
export const FRAGMENTS_EDITOR_ITEM_BORDERS = {
	bottom: 'border-bottom',
	top: 'border-top'
};

/**
 * Fragments Editor item types
 * @review
 * @type {!object}
 */
export const FRAGMENTS_EDITOR_ITEM_TYPES = {
	column: 'fragments-editor-column',
	fragment: 'fragments-editor-fragment',
	fragmentList: 'fragments-editor-fragment-list',
	section: 'fragments-editor-section'
};