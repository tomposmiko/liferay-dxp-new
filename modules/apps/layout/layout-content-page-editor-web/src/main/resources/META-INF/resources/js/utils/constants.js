/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

/**
 * @type {string}
 */
export const BACKGROUND_IMAGE_FRAGMENT_ENTRY_PROCESSOR =
	'com.liferay.fragment.entry.processor.background.image.BackgroundImageFragmentEntryProcessor';

/**
 * List of editable types and their compatibilities
 * with the corresponding mappeable types
 * @review
 * @see DDMStructureClassType.java for compatible types
 * @type {!object}
 */

export const COMPATIBLE_TYPES = {
	html: [
		'ddm-date',
		'ddm-decimal',
		'ddm-integer',
		'ddm-number',
		'ddm-text-html',
		'text',
		'textarea',
		'url'
	],

	image: ['ddm-image', 'image'],

	link: [
		'ddm-date',
		'ddm-decimal',
		'ddm-integer',
		'ddm-number',
		'ddm-text-html',
		'text',
		'textarea',
		'url'
	],

	'rich-text': [
		'ddm-date',
		'ddm-decimal',
		'ddm-integer',
		'ddm-number',
		'ddm-text-html',
		'text',
		'textarea',
		'url'
	],

	text: [
		'ddm-date',
		'ddm-decimal',
		'ddm-integer',
		'ddm-number',
		'text',
		'textarea',
		'url'
	]
};

/**
 * Event types for creating the text editor processor
 */
export const CREATE_PROCESSOR_EVENT_TYPES = {
	button: 'button',
	editable: 'editable'
};

/**
 * Available editable field config keys
 */
export const EDITABLE_FIELD_CONFIG_KEYS = {
	alt: 'alt',
	imageLink: 'imageLink',
	imageSource: 'imageSource',
	imageTarget: 'imageTarget',
	imageTitle: 'imageTitle',
	textAlignment: 'textAlignment',
	textColor: 'textColor',
	textStyle: 'textStyle'
};

/**
 * FloatingToolbar panels
 */
export const FLOATING_TOOLBAR_BUTTONS = {
	backgroundColor: {
		icon: 'color-picker',
		id: 'background_color',
		panelId: 'background_color',
		title: Liferay.Language.get('background-color'),
		type: 'panel'
	},

	duplicateFragment: {
		icon: 'paste',
		id: 'duplicate_fragment',
		panelId: 'duplicate_fragment',
		title: Liferay.Language.get('duplicate-fragment'),
		type: 'panel'
	},

	edit: {
		icon: 'pencil',
		id: 'edit',
		title: Liferay.Language.get('edit'),
		type: 'editor'
	},

	fragmentBackgroundImage: {
		icon: 'pencil',
		id: 'fragment_background_image',
		panelId: 'fragment_background_image',
		title: Liferay.Language.get('fragment-background-image'),
		type: 'panel'
	},

	fragmentConfiguration: {
		icon: 'cog',
		id: 'fragment_configuration',
		panelId: 'fragment_configuration',
		title: Liferay.Language.get('fragment-configuration'),
		type: 'panel'
	},

	imageProperties: {
		icon: 'pencil',
		id: 'image_properties',
		panelId: 'image_properties',
		title: Liferay.Language.get('image-properties'),
		type: 'panel'
	},

	layoutBackgroundImage: {
		icon: 'picture',
		id: 'layout_background_image',
		panelId: 'layout_background_image',
		title: Liferay.Language.get('layout-background-image'),
		type: 'panel'
	},

	link: {
		icon: 'link',
		id: 'link',
		panelId: 'link',
		title: Liferay.Language.get('link'),
		type: 'panel'
	},

	map: {
		icon: 'bolt',
		id: 'mapping',
		panelId: 'mapping',
		title: Liferay.Language.get('map'),
		type: 'panel'
	},

	spacing: {
		icon: 'table',
		id: 'spacing',
		panelId: 'spacing',
		title: Liferay.Language.get('spacing'),
		type: 'panel'
	}
};

/**
 * @type {string}
 */
export const EDITABLE_FRAGMENT_ENTRY_PROCESSOR =
	'com.liferay.fragment.entry.processor.editable.EditableFragmentEntryProcessor';

/**
 * Fragments Editor dragging class
 * @review
 * @type {string}
 */
export const FRAGMENTS_EDITOR_DRAGGING_CLASS = 'dragging';

/**
 * Fragments Editor item borders
 * @review
 * @type {!object}
 */
export const FRAGMENTS_EDITOR_ITEM_BORDERS = {
	bottom: 'fragments-editor-border-bottom',
	top: 'fragments-editor-border-top'
};

/**
 * Fragments Editor item types
 * @review
 * @type {!object}
 */
export const FRAGMENTS_EDITOR_ITEM_TYPES = {
	backgroundImageEditable: 'fragments-editor-background-image-editable',
	column: 'fragments-editor-column',
	editable: 'fragments-editor-editable-field',
	fragment: 'fragments-editor-fragment',
	fragmentList: 'fragments-editor-fragment-list',
	mappedItem: 'fragments-editor-mapped-item',
	row: 'fragments-editor-row'
};

/**
 * Fragments Editor row types
 * @review
 * @type {!object}
 */
export const FRAGMENTS_EDITOR_ROW_TYPES = {
	componentRow: '1',
	sectionRow: '0'
};

/**
 * @type {string}
 */
export const FREEMARKER_FRAGMENT_ENTRY_PROCESSOR =
	'com.liferay.fragment.entry.processor.freemarker.FreeMarkerFragmentEntryProcessor';

/**
 * Available mapping source type ids
 */
export const MAPPING_SOURCE_TYPE_IDS = {
	content: 'specific_content',
	structure: 'structure'
};

/**
 * Available attributes for target config key
 */
export const TARGET_TYPES = [
	{
		label: Liferay.Language.get('self'),
		targetTypeId: '_self'
	},
	{
		label: Liferay.Language.get('blank'),
		targetTypeId: '_blank'
	},
	{
		label: Liferay.Language.get('parent'),
		targetTypeId: '_parent'
	},
	{
		label: Liferay.Language.get('top'),
		targetTypeId: '_top'
	}
];
