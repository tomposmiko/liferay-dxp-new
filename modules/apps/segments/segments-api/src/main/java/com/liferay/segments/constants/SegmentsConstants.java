/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.segments.constants;

import com.liferay.portal.kernel.language.LanguageUtil;
import com.liferay.portal.kernel.util.ResourceBundleUtil;

import java.util.Locale;
import java.util.ResourceBundle;

/**
 * @author Eduardo Garc√≠a
 */
public class SegmentsConstants {

	public static final String RESOURCE_NAME = "com.liferay.segments";

	/**
	 * @deprecated As of Mueller (7.2.x)
	 */
	@Deprecated
	public static final long SEGMENTS_ENTRY_ID_DEFAULT = 0;

	/**
	 * @deprecated As of Mueller (7.2.x)
	 */
	@Deprecated
	public static final long SEGMENTS_EXPERIENCE_ID_DEFAULT = 0;

	/**
	 * @deprecated As of Mueller (7.2.x)
	 */
	@Deprecated
	public static final String SEGMENTS_EXPERIENCE_ID_PREFIX =
		"segments-experience-id-";

	/**
	 * @deprecated As of Mueller (7.2.x)
	 */
	@Deprecated
	public static final int SEGMENTS_EXPERIENCE_PRIORITY_DEFAULT = -1;

	public static final String SERVICE_NAME = "com.liferay.segments";

	/**
	 * @deprecated As of Mueller (7.2.x)
	 */
	@Deprecated
	public static final String SOURCE_ASAH_FARO_BACKEND = "ASAH_FARO_BACKEND";

	/**
	 * @deprecated As of Mueller (7.2.x)
	 */
	@Deprecated
	public static final String SOURCE_DEFAULT = "DEFAULT";

	/**
	 * @deprecated As of Mueller (7.2.x)
	 */
	@Deprecated
	public static String getDefaultSegmentsEntryName(Locale locale) {
		ResourceBundle resourceBundle = ResourceBundleUtil.getBundle(
			"content.Language", locale, SegmentsConstants.class);

		return LanguageUtil.get(resourceBundle, "default-segment-name");
	}

	/**
	 * @deprecated As of Mueller (7.2.x)
	 */
	@Deprecated
	public static String getDefaultSegmentsExperienceName(Locale locale) {
		ResourceBundle resourceBundle = ResourceBundleUtil.getBundle(
			"content.Language", locale, SegmentsConstants.class);

		return LanguageUtil.get(resourceBundle, "default-experience-name");
	}

}