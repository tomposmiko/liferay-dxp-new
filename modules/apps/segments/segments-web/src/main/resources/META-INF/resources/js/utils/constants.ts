/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

/**
 * Constants for OData query.
 */

import {sub} from 'frontend-js-web';

export const CONJUNCTIONS = {
	AND: 'and',
	OR: 'or',
} as const;

export const FUNCTIONAL_OPERATORS = {
	CONTAINS: 'contains',
} as const;

export const NOT_OPERATORS = {
	NOT_CONTAINS: 'not-contains',
	NOT_EQ: 'not-eq',
} as const;

export const HAS_OPERATORS = {
	HAS: 'has',
	NOT_HAS: 'has-not',
} as const;

export const RELATIONAL_OPERATORS = {
	EQ: 'eq',
	GE: 'ge',
	GT: 'gt',
	LE: 'le',
	LT: 'lt',
} as const;

export const DATE_OPERATORS = {
	BETWEEN: 'between',
	EVER: 'ever',
} as const;

export const SINCE_VALUES = {
	last7Days: sub(Liferay.Language.get('last-x-days'), 7),
	last24Hours: sub(Liferay.Language.get('last-x-hours'), 24),
	last28Days: sub(Liferay.Language.get('last-x-days'), 28),
	last30Days: sub(Liferay.Language.get('last-x-days'), 30),
	last90Days: sub(Liferay.Language.get('last-x-days'), 90),
	yesterday: Liferay.Language.get('yesterday'),
} as const;

/**
 * Constants to match property types in the passed in supportedProperties array.
 */

export const PROPERTY_TYPES = {
	BOOLEAN: 'boolean',
	COLLECTION: 'collection',
	DATE: 'date',
	DATE_TIME: 'date-time',
	DOUBLE: 'double',
	EVENT: 'event',
	ID: 'id',
	INTEGER: 'integer',
	STRING: 'string',
} as const;

export const PROPERTY_GROUPS = {
	EVENT: 'event',
} as const;

/**
 * Constants for CriteriaBuilder component.
 */

const {AND, OR} = CONJUNCTIONS;
const {EQ, GE, GT, LE, LT} = RELATIONAL_OPERATORS;
const {NOT_CONTAINS, NOT_EQ} = NOT_OPERATORS;
const {HAS, NOT_HAS} = HAS_OPERATORS;
const {CONTAINS} = FUNCTIONAL_OPERATORS;
const {
	BOOLEAN,
	COLLECTION,
	DATE,
	DATE_TIME,
	DOUBLE,
	EVENT,
	ID,
	INTEGER,
	STRING,
} = PROPERTY_TYPES;
const {BETWEEN, EVER} = DATE_OPERATORS;
const {
	last7Days,
	last24Hours,
	last28Days,
	last30Days,
	last90Days,
	yesterday,
} = SINCE_VALUES;

export const SUPPORTED_CONJUNCTIONS = [
	{
		label: Liferay.Language.get('and'),
		name: AND,
	},
	{
		label: Liferay.Language.get('or'),
		name: OR,
	},
] as const;

export const SUPPORTED_OPERATORS = [
	{
		label: Liferay.Language.get('equals'),
		name: EQ,
	},
	{
		label: Liferay.Language.get('not-equals'),
		name: NOT_EQ,
	},
	{
		label: Liferay.Language.get('greater-than'),
		name: GT,
	},
	{
		label: Liferay.Language.get('greater-than-or-equals'),
		name: GE,
	},
	{
		label: Liferay.Language.get('less-than'),
		name: LT,
	},
	{
		label: Liferay.Language.get('less-than-or-equals'),
		name: LE,
	},
	{
		label: Liferay.Language.get('contains'),
		name: CONTAINS,
	},
	{
		label: Liferay.Language.get('not-contains'),
		name: NOT_CONTAINS,
	},
] as const;

export const SUPPORTED_EVENT_OPERATORS = [
	{
		label: Liferay.Language.get('at-least'),
		name: GE,
	},
	{
		label: Liferay.Language.get('at-most'),
		name: LE,
	},
	{
		label: Liferay.Language.get('has'),
		name: HAS,
	},
	{
		label: Liferay.Language.get('has-not'),
		name: NOT_HAS,
	},
] as const;

export const SUPPORTED_EVENT_DATE_OPERATORS = [
	{
		label: Liferay.Language.get('between'),
		name: BETWEEN,
	},
	{
		label: Liferay.Language.get('on'),
		name: EQ,
	},
	{
		label: Liferay.Language.get('ever'),
		name: EVER,
	},
	{
		label: Liferay.Language.get('after'),
		name: GT,
	},
	{
		label: Liferay.Language.get('before'),
		name: LT,
	},
] as const;

export const SUPPORTED_PROPERTY_TYPES = {
	[BOOLEAN]: [EQ, NOT_EQ],
	[COLLECTION]: [EQ, NOT_EQ, CONTAINS, NOT_CONTAINS],
	[DATE]: [EQ, GE, GT, LE, LT, NOT_EQ],
	[DATE_TIME]: [EQ, GE, GT, LE, LT, NOT_EQ],
	[DOUBLE]: [EQ, GE, GT, LE, LT, NOT_EQ],
	[EVENT]: {
		DATE: [BETWEEN, EQ, EVER, GT, LT],
		INTEGER: [GE, LE],
		NOT: [HAS, NOT_HAS],
		SINCE: [
			last7Days,
			last24Hours,
			last28Days,
			last30Days,
			last90Days,
			yesterday,
		],
	},
	[ID]: [EQ, NOT_EQ],
	[INTEGER]: [EQ, GE, GT, LE, LT, NOT_EQ],
	[STRING]: [EQ, NOT_EQ, CONTAINS, NOT_CONTAINS],
} as const;

/**
 * Values for criteria row inputs.
 */

export const BOOLEAN_OPTIONS = [
	{
		label: Liferay.Language.get('true'),
		value: 'true',
	},
	{
		label: Liferay.Language.get('false'),
		value: 'false',
	},
] as const;
