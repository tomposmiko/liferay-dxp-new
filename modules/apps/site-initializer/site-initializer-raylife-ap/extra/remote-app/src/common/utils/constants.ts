/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

export type ConstantListType = {[keys: string]: string};

type ConstantsType = {
	APPLICATION_STATUS: {[keys: string]: {INDEX: number; NAME: string}};
	DEVICES: ConstantListType;
	MONTHS_ABREVIATIONS: string[];
	US_STATES: ConstantListType[];
};

export const CONSTANTS: ConstantsType = {
	APPLICATION_STATUS: {
		approved: {INDEX: 8, NAME: 'approved'},
		bound: {INDEX: 6, NAME: 'bound'},
		inInvestigation: {INDEX: 7, NAME: 'InInvestigation'},
		incomplete: {INDEX: 1, NAME: 'incomplete'},
		open: {INDEX: 0, NAME: 'open'},
		quoted: {INDEX: 2, NAME: 'quoted'},
		rejected: {INDEX: 5, NAME: 'rejected'},
		reviewed: {INDEX: 4, NAME: 'reviewed'},
		underwriting: {INDEX: 3, NAME: 'underwriting'},
	},

	DEVICES: {
		DESKTOP: 'DESKTOP',
		PHONE: 'PHONE',
		TABLET: 'TABLET',
	},

	MONTHS_ABREVIATIONS: [
		'Jan',
		'Feb',
		'Mar',
		'Apr',
		'May',
		'Jun',
		'Jul',
		'Aug',
		'Sep',
		'Oct',
		'Nov',
		'Dec',
	],

	US_STATES: [
		{
			label: '',
			value: '',
		},
		{
			label: 'CHOOSE AN OPTION',
			value: 'CHOOSE AN OPTION',
		},
		{
			label: 'CA',
			value: 'CA',
		},
		{
			label: 'NV',
			value: 'NV',
		},
		{
			label: 'NY',
			value: 'NY',
		},
	],
};

export const productAutoERC = 'RAYAP-001';
