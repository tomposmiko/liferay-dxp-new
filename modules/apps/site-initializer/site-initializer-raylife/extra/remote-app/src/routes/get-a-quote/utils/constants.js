export const TOTAL_OF_FIELD = {
	BASICS: 10,
	EMPLOYEES: 7,
	PROPERTY: 6,
};

export const AVAILABLE_STEPS = {
	BASICS_BUSINESS_INFORMATION: {
		section: 'basics',
		subsection: 'businessInformation',
		title: 'Just the business basics!',
	},
	BASICS_BUSINESS_TYPE: {
		section: 'basics',
		subsection: 'business-type',
		title: "Welcome! Let's start.",
	},
	BASICS_PRODUCT_QUOTE: {
		section: 'basics',
		subsection: 'product-quote',
		title: 'Select a product.',
	},
	BUSINESS: {
		section: 'business',
		subsection: '',
		title: "Let's get to know your business!",
	},
	EMPLOYEES: {
		section: 'employees',
		subsection: '',
		title: 'Tell us about your employees!',
	},
	PROPERTY: {
		section: 'property',
		subsection: '',
		title: 'More about',
	},
};
