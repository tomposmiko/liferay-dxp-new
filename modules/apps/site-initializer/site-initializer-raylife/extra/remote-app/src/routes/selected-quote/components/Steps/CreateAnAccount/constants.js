export const NATURAL_VALUE = 0;
export const CHECK_VALUE = 1;
export const UNCHECKED_VALUE = 2;

export const INITIAL_VALIDATION = {
	noSpace: NATURAL_VALUE,
	numberCharacter: NATURAL_VALUE,
	qtdCharacter: NATURAL_VALUE,
	samePassword: NATURAL_VALUE,
	symbolCharacter: NATURAL_VALUE,
	uniqueCharacter: NATURAL_VALUE,
};
