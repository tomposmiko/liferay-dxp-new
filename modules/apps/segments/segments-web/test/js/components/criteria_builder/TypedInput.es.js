import React from 'react';
import TypedInput from 'components/criteria_builder/TypedInput.es';
import {PROPERTY_TYPES} from 'utils/constants.es';
import {cleanup, fireEvent, render} from 'react-testing-library';

const DATE_INPUT_TESTID = 'date-input';

const DECIMAL_NUMBER_INPUT_TESTID = 'decimal-number';

const ENTITY_SELECT_INPUT_TESTID = 'entity-select-input';

const INTEGER_NUMBER_INPUT_TESTID = 'integer-number';

const OPTIONS_BOOLEAN_INPUT_TESTID = 'options-boolean';

const OPTIONS_STRING_INPUT_TESTID = 'options-string';

const SIMPLE_STRING_INPUT_TESTID = 'simple-string';

const defaultValue = 'defaultValue';

function testControlledInput(
	{
		element,
		mockFunc,
		newValue = 'Liferay',
		newValueExpected,
		value
	}
) {
	expect(element.value).toBe(value);

	fireEvent.change(
		element,
		{
			target: {value: newValue}
		}
	);

	expect(mockFunc.mock.calls.length).toBe(1);
	expect(mockFunc.mock.calls[0][0]).toBe(newValueExpected || newValue);

	expect(element.value).toBe(value);
}

describe(
	'CriteriaRow',
	() => {
		afterEach(cleanup);

		it(
			'should render with type string',
			() => {
				const {asFragment} = render(
					<TypedInput
						onChange={jest.fn()}
						type="string"
					/>
				);

				expect(asFragment()).toMatchSnapshot();
			}
		);

		it(
			'should render type string with value',
			() => {
				const mockOnChange = jest.fn();

				const {asFragment, getByTestId} = render(
					<TypedInput
						onChange={mockOnChange}
						type="string"
						value={defaultValue}
					/>
				);

				expect(asFragment()).toMatchSnapshot();

				const element = getByTestId(SIMPLE_STRING_INPUT_TESTID);

				testControlledInput(
					{
						element,
						mockFunc: mockOnChange,
						value: defaultValue
					}
				);
			}
		);

		it(
			'should render type string with options',
			() => {
				const mockOnChange = jest.fn();

				const options = [
					{
						label: 'Default Value',
						value: 'defaultValue'
					}, {
						label: 'LIFERAY',
						value: 'Liferay'
					}
				];

				const {asFragment, getByTestId} = render(
					<TypedInput
						onChange={mockOnChange}
						options={options}
						type="string"
						value={defaultValue}
					/>
				);

				expect(asFragment()).toMatchSnapshot();

				const element = getByTestId(OPTIONS_STRING_INPUT_TESTID);

				testControlledInput(
					{
						element,
						mockFunc: mockOnChange,
						value: defaultValue
					}
				);
			}
		);

		it(
			'should render type boolean',
			() => {
				const mockOnChange = jest.fn();

				const defaultBoolValue = 'true';

				const {asFragment, getByTestId} = render(
					<TypedInput
						onChange={mockOnChange}
						type={PROPERTY_TYPES.BOOLEAN}
						value={defaultBoolValue}
					/>
				);

				expect(asFragment()).toMatchSnapshot();

				const element = getByTestId(OPTIONS_BOOLEAN_INPUT_TESTID);

				testControlledInput(
					{
						element,
						mockFunc: mockOnChange,
						newValue: 'false',
						value: defaultBoolValue
					}
				);
			}
		);

		it(
			'should render type integer number',
			() => {
				const mockOnChange = jest.fn();

				const defaultNumberValue = '1';

				const {asFragment, getByTestId} = render(
					<TypedInput
						onChange={mockOnChange}
						type={PROPERTY_TYPES.INTEGER}
						value={defaultNumberValue}
					/>
				);

				expect(asFragment()).toMatchSnapshot();

				const element = getByTestId(INTEGER_NUMBER_INPUT_TESTID);

				testControlledInput(
					{
						element,
						mockFunc: mockOnChange,
						newValue: '2',
						value: defaultNumberValue
					}
				);
			}
		);

		it(
			'should render type decimal number',
			() => {
				const mockOnChange = jest.fn();

				const defaultNumberValue = '1.23';
				const newNumberValue = '2.34';

				const {asFragment, getByTestId} = render(
					<TypedInput
						onChange={mockOnChange}
						type={PROPERTY_TYPES.DOUBLE}
						value={defaultNumberValue}
					/>
				);

				expect(asFragment()).toMatchSnapshot();

				const element = getByTestId(DECIMAL_NUMBER_INPUT_TESTID);

				// Needs to be tested a different way from other inputs since
				// the onChange is called in onBlur rather than on input change.

				expect(element.value).toBe(defaultNumberValue);

				fireEvent.change(
					element,
					{
						target: {value: newNumberValue}
					}
				);

				fireEvent.blur(element);

				expect(element.value).toBe(newNumberValue);
			}
		);

		it(
			'should render type date',
			() => {
				const mockOnChange = jest.fn();

				const defaultNumberValue = '2019-01-23';

				const {asFragment, getByTestId} = render(
					<TypedInput
						onChange={mockOnChange}
						type={PROPERTY_TYPES.DATE}
						value={defaultNumberValue}
					/>
				);

				expect(asFragment()).toMatchSnapshot();

				const element = getByTestId(DATE_INPUT_TESTID);

				testControlledInput(
					{
						element,
						mockFunc: mockOnChange,
						newValue: '2019-01-24',
						newValueExpected: '2019-01-24T00:00:00.000Z',
						value: defaultNumberValue
					}
				);
			}
		);

		it(
			'should render type id',
			() => {
				const mockOnChange = jest.fn();

				const defaultNumberValue = '12345';

				const {getByTestId} = render(
					<TypedInput
						onChange={mockOnChange}
						selectEntity={{
							id: 'entitySelect',
							title: 'Select Entity Test',
							url: ''
						}}
						type={PROPERTY_TYPES.ID}
						value={defaultNumberValue}
					/>
				);

				const element = getByTestId(ENTITY_SELECT_INPUT_TESTID);

				expect(element.value).toBe(defaultNumberValue);
			}
		);
	}
);