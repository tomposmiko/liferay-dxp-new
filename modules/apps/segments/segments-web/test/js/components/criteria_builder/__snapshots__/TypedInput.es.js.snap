// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CriteriaRow should render type boolean 1`] = `
<DocumentFragment>
  <select
    class="form-control criterion-input form-control"
    data-testid="options-boolean"
  >
    <option
      value="true"
    >
      TRUE
    </option>
    <option
      value="false"
    >
      FALSE
    </option>
  </select>
</DocumentFragment>
`;

exports[`CriteriaRow should render type date 1`] = `
<DocumentFragment>
  <div
    class="criterion-input date-input"
  >
    <input
      class="form-control"
      data-testid="date-input"
      type="date"
      value="2019-01-23"
    />
  </div>
</DocumentFragment>
`;

exports[`CriteriaRow should render type decimal number 1`] = `
<DocumentFragment>
  <input
    class="criterion-input form-control"
    data-testid="decimal-number"
    step="0.01"
    type="number"
    value="1.23"
  />
</DocumentFragment>
`;

exports[`CriteriaRow should render type integer number 1`] = `
<DocumentFragment>
  <input
    class="criterion-input form-control"
    data-testid="integer-number"
    type="number"
    value="1"
  />
</DocumentFragment>
`;

exports[`CriteriaRow should render type string with options 1`] = `
<DocumentFragment>
  <select
    class="form-control criterion-input form-control"
    data-testid="options-string"
  >
    <option
      value="defaultValue"
    >
      Default Value
    </option>
    <option
      value="Liferay"
    >
      LIFERAY
    </option>
  </select>
</DocumentFragment>
`;

exports[`CriteriaRow should render type string with value 1`] = `
<DocumentFragment>
  <input
    class="criterion-input form-control"
    data-testid="simple-string"
    type="text"
    value="defaultValue"
  />
</DocumentFragment>
`;

exports[`CriteriaRow should render with type string 1`] = `
<DocumentFragment>
  <input
    class="criterion-input form-control"
    data-testid="simple-string"
    type="text"
    value=""
  />
</DocumentFragment>
`;
